packages/notifications/push/amplify_push_notifications/lib/src/amplify_push_notifications_impl.dart

```diff
@@ -335,14 +335,26 @@ abstract class AmplifyPushNotifications
 
   Future<void> _registerDeviceWhenConfigure() async {
     late String deviceToken;
+
     try {
-      deviceToken = await _bufferedTokenStream.peek;
+      await _hostApi.requestInitialToken();
+      deviceToken =
+          await _bufferedTokenStream.peek.timeout(const Duration(seconds: 2));
```

Comment:
Is 2 seconds long enough? Can we make it 5 or 10 to be safe?

Reply:
On iOS and Android token generation is client behavior, and doesn't require network round trip... it should be quick. But can increase the timeout surely to avoid surprises, as this timeout is for catching potential error without increasing latency for normal use cases. Will update to 5.

Reply:
Makes sense. Thanks!